source/compiler/qsc_circuit/src/builder.rs

```diff
@@ -42,6 +42,7 @@ pub struct CircuitTracer {
     circuit_builder: OperationListBuilder,
     next_result_id: usize,
     user_package_ids: Vec<PackageId>,
+    nontrimmable_qubits: FxHashSet<usize>,
```

Comment:
A more descriptive name, such as `used_qubits`, `necessary_qubits`, `qubits_in_superposition` or `qubits_ever_in_superposition`?

(If you have to stick with `nontrimmable` - at least `untrimmable` sounds more natural to my ears)

Reply:
I went with `superposition_qubits`